velox/common/base/SimdUtil.h

```diff
@@ -480,8 +480,7 @@ inline void storeLeading(
 
 /// Stores elements of 'input' selected by 'indices' into 'output'. output[i] =
 /// input[indices[i]].
-/// Indices and output may be the same. May over-read indices but will not
-/// dereference indices that are not in range. Writes exactly indices.size()
+/// Indices and output may be the same. Writes exactly indices.size()
```

Comment:
Can you make a different version that does not overread if you need it?  We pad the buffer so in most of cases overread is fine, and you fix has impact on performance, so should be an opt in.